File: Python/7py 1/four_seven.py
"""
|--------------------------------------------------------------------------
| Four/Seven
|--------------------------------------------------------------------------
|
"""

def solution(n):
   return {7: 4, 4: 7}.get(n, 0)


solution = {4:7, 7:4}.get


def solution(n):
   while n==7: return 4
   while n==4: return 7
   return False


print(solution(4), 7)
print(solution(7), 4)

File: PHP/7php 1/fourSeven.php
<?php
/*
|--------------------------------------------------------------------------
| Four/Seven
|--------------------------------------------------------------------------
|
*/

function solution($n) {
   return 4*($n==7) + 7*($n==4);
}


function solution($n) {
   return [4 => 7, 7 => 4][$n];
}


function solution($n) {
   while ($n == 4 || $n == 7) return $n ^ 3;
   return 0;
}


print_r(solution(4) . ' ' . 7 . '
');
print_r(solution(7) . ' ' . 4 . '
');